### DISCUSSION TITLE
WSA 2305 Update!

### DISCUSSION THREAD

**Discussion by hamza-usmani** [microsoft/WSA#322] [Announcements]
# What's New

We've shipped an update for Windows Subsystem for Android™ on Windows 11 to the [Windows Subsystem for Android™ Preview Program](https://learn.microsoft.com/en-us/windows/android/wsa/preview-program). This update (2305.40000.2.0) will add the long-requested **file sharing**, settings updates and graphics fixes.

## File Sharing
Apps can now directly access files in the Windows filesystem through the Windows known folders (See our detailed post about file sharing: https://github.com/microsoft/WSA/discussions/321!

## Full Release Notes
- File sharing enabled
- File transfer using drop and drop and copy and paste!
- Windows Subsystem for Android Settings (renamed to just "Windows Subsystem for Android") redesign, including displaying all installed Android apps
- Enable apps that specify android.hardware.type.pc in their manifest to opt into receiving raw input events 
- Wi-Fi API compatibility improvements
- Camera hardware compatibility improvements
- Linux kernel security update
- Updated latest Chromium WebView to version 113
- Android 13 security updates

# Giving feedback

If you are having issues with Windows Subsystem for Android™ – please file feedback via Feedback Hub under Apps > Windows Subsystem for Android™. For more information about troubleshooting and submitting feedback: [Troubleshooting and FAQ for mobile apps on Windows](https://support.microsoft.com/en-us/windows/troubleshooting-and-faq-for-mobile-apps-on-windows-d6062afd-98a1-4018-a7c8-6b3b680a2ea5).

Thanks,
Windows Subsystem for Android™ Team

**Comment by HotCakeX** [comment]
New features ❤️‍🔥

![image](https://github.com/microsoft/WSA/assets/118815227/534dc090-3130-45d4-a744-aafbe7db4770)

**Comment by 7gxycn08** [comment]
Which channel is the WSA 2305 Update pushed to? Can't find it.

**Reply by HotCakeX** [reply]
I'm using Windows 11 insider beta channel and I received this WSA update automatically.

**Reply by MustardChef** [reply]
> Which channel is the WSA 2305 Update pushed to? Can't find it.

Sign up for the WSA Preview Program: https://aka.ms/WSAPreviewProgram

**Reply by jo-chemla** [reply]
I've signed up a few weeks ago (using the same email than that of the account used in Microsoft Store) but my WSA is still locked at 2304. Any ay to force WSA to update? Thanks!

**Reply by MustardChef** [reply]
@jo-chemla Check for updates in the Microsoft Store. If you are not still not in the Preview Program, try reapplying through the above link. 

But if you really want to update without waiting, you can check out my repo: https://github.com/MustardChef/WSAPackages

**Reply by jo-chemla** [reply]
Yep Microsoft Store says WSA app is up to date, and I already tried reapplying a bunch of times to the preview program in the past few weeks. Will see if anything changes in the coming days, otherwise I might as well try out your build (although it would involve resetting WSA from scratch). Thanks for the suggestion!

**Reply by MustardChef** [reply]
@jo-chemla You should be able to update to 2305 without losing data, if you download the .MSIX bundle from the link that I provided. Try it.

**Comment by upintheairsheep** [comment]
Can you make some sort of command line flag to allow insecure extensions, such as .exe to be transferred?

**Reply by HotCakeX** [reply]
What would be the point? Android doesn't understand .exe and that feature can be a security risk, potentially bypassing SmartScreen specially when browsers that users install on WSA can act as system-wide browser for the whole system (including host).

**Comment by Bob3500** [comment]
Dear WSA team,
With the 2305 update, the displaying of installed apps has appeared but the compatibility entry has disappeared. this entry was providing compatibility settings for each app (see screenshot below)

![image](https://github.com/microsoft/WSA/assets/137118151/3749654a-c69b-4074-996d-4ac33dbd8452)

On my side, I need the ability to force automatic resizing of my app (see the screenshot below when automatic resize is not set)

![image](https://github.com/microsoft/WSA/assets/137118151/51ba00bf-c376-4f5d-8399-b2c4ce32918a)

From the release 2305, I don't find anymore the way to force automatic resizing.

Would you have any suggestion for this ?

Thank you for your support.
Best regards.

**Reply by MustardChef** [reply]
@Bob3500 I also have realised that the the compatibility menu has disappared, but it seems that this move is intentional.
To let the WSA Development team know that you found the Compatibility Menu useful and which features that you have found useful, here is the recommended course of action:
![image](https://github.com/microsoft/WSA/assets/68516357/5533dc8e-994a-4b1a-b341-964af0eead66)

**Reply by Bob3500** [reply]
Thank you @MustardChef. Yes I also push a feedback from the Windows Feedback Hub !
Maybe someone has a suggestion on the Android app development side, to explicitely indicate that the app is resizable (app manifest ? other ? ...).

**Comment by omidsky2600** [comment]
The windows subsystem for android program occupies a large amount of RAM and CPU memory
If the system RAM is 16 GB, it takes 12 GB of memory
It crashes by itself

**Reply by HotCakeX** [reply]
No it doesn't, I have exactly 16GB of RAM and I've been using it since day 1.

**Comment by Cashew-wood** [comment]
There is a Memory leak problem. The computer's 16G memory wsa uses 6G memory to browse the "Tiktok" wsa for two hours. The wsa crashes directly and automatically restarts, sometimes resulting in a black screen on the computer.

**Comment by omidsky2600** [comment]
![IMG_20230630_034917](https://github.com/microsoft/WSA/assets/106124109/308d9357-1c7b-4a7e-9411-5ef33f86c378)
why doesn't it show the keyboard when  l installed  the keyboard  app ?
I installed. The. Google  keyboard. But it does. Not. Show the letters
Every keyboard program I install is the same
There was no such problem before when there was Android 12
This happened when Android 13 was released
